blog/_posts/2021-11-12-ahed-project.md

```diff
@@ -0,0 +1,392 @@
+
+titles_matched = pd.DataFrame(data, columns = ['bibcode','TITLE'])
+
+```
+<b>Step 3.3: Match Bibcode/Title Response Data to Original Paper List</b>
+  
+Finally, I was able to merge the new titles & bibcodes to my running list of matches ('refs_matched').
+  
+```python
+# Merge/Join new table to original, joined on 'TITLE'
+merged = df.merge(titles_matched, on='TITLE', how='left')
+
+# Combine bibcode columns
+merged['BIBCODE'] = merged['BIBCODE'].fillna(merged['bibcode'])
+merged = merged.drop('bibcode',axis=1)
+
+# Clean up nulls
+merged = merged.replace(np.nan,'NA')
+
+# Export merged data to new excel file
+merged.to_excel("AHED/final_matched_2.xlsx", index=False)
+```
+With this merge, my total came up to about 692 items matched out of a potential 797. After some analysis of what was left unmatched, I found quite a few discrepancies in the 'Year' metadata, as well as 'Title' mismatches (i.e. typos in the metadata, titles changed during publication, etc.) so when I searched again by Title alone, I ended up finding an additional ~40 or so papers that matched ADS holdings, bringing my final total to 731 items.
+
+  ## <a name="ads-libs">Task 4 (Notebook 4): Curate missing items and create ADS Libraries</a>
+
+After successfully identifying as many bibcodes as I could match between AHED and ADS, I grabbed my full list of bibcodes and I made another ADS Library of them, which can be accessed [here](https://ui.adsabs.harvard.edu/user/libraries/1gM2Y7nVSv-POu2lanjJ6g). 
+  
+My final task for this project was to identify, locate, and curate the last ~70 AHED records missing from the ADS holdings. This was a manual process of searching the web with the metadata provided, locating the applicable DOI, and curating ADS records from there. I was able to identify and locate approximately 54 publications, plus records for individual chapters for two I identified as books. As a result, the ADS team ingested these new records and I again created a library of those (102 total), which can be found [here](https://ui.adsabs.harvard.edu/user/libraries/HkCPGwYhSSWpzvJW_gxd3w).
+  
+```python
+import requests
+import json
+
+# --- API REQUEST --- 
+token = "<my token here>"
+url = "https://api.adsabs.harvard.edu/v1/biblib/libraries"
+    
+data = { 
+    "name":"AHED Library",
+    "description":"Library of records ADS matches of AHED holdings",
+    "public": True,
+    "bibcode": bibs
+}
+headers = {'Authorization': 'Bearer ' + token}
+response = requests.post(url, data=json.dumps(data), headers=headers)
+
+print(response.status_code)
+```
+
+    200
+
+## Lessons Learned
+- Using OpenRefine to do the bulk of your data prep and cleanup was a great advantage
+- Try to save fewer files; Exporting/importing new csvs/files with small tweaks can disrupt the flow of the script and create a bunch of extra files that will clutter your documents.
+- Look for more Python ways of measuring my progress - how many bibcodes matched, counting duplicates vs non-duplicates
```

Comment:
I'd change "Python" to "quantitative", since I think that's really where you're going here